jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/migration/MigrationManager.java

```diff
@@ -592,6 +606,15 @@ public Void execute(Context context) {
                             for (TimerInstance timerInstance : timerInstances) {
                                 long delay = timerInstance.getDelay() - (System.currentTimeMillis() - timerInstance.getActivated().getTime());
                                 timerInstance.setDelay(delay);
+                                
+                                Trigger trigger = ((DefaultJobHandle)timerInstance.getJobHandle()).getTimerJobInstance().getTrigger();
+                                if (trigger instanceof IntervalTrigger && timerInstance.getPeriod() > 0) {
+                                    if (timerInstance.getRepeatLimit() > 0) {
+                                        timerInstance.setRepeatLimit(((IntervalTrigger) trigger).getRepeatLimit() - ((IntervalTrigger) trigger).getRepeatCount());
+                                    }
+                                    delay = timerInstance.getPeriod() - (System.currentTimeMillis() - ((IntervalTrigger) trigger).getLastFireTime().getTime());
+                                    timerInstance.setDelay(delay);
+                                }
```

Comment:
I think we could extract this to a method, wdyt?

Reply:
extracted to method